src/components/ui/Chat/MessageList/DateHeader.tsx

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { HTMLAttributes } from 'react';
+
+import { StyledDateHeader } from './StyledDateHeader';
+import { BaseProps } from '../../Base';
+
+export interface DateHeaderProps extends Omit<HTMLAttributes<HTMLDivElement>, 'css'>, BaseProps {
+  date: string;
+  key: string;
+}
+
+export const DateHeader = (props: DateHeaderProps) => {
+  const dateOptions = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
+
+  const formatDate = (dateStr: string) => {
+    const date = new Date(dateStr).toLocaleDateString(undefined, dateOptions)
+    const today = new Date().toLocaleDateString(undefined, dateOptions);
+    const yesterday = new Date(Date.now() - 24 * 60 * 60 * 1000).toLocaleDateString(undefined, dateOptions)
+    if (date === yesterday) {
+      return 'Yesterday'
+    } else if (date === today) {
+      return 'Today'
+    }
+    return (
+      date
+    )
+  }
+
```

Comment:
I'd recommend a few things here:
* This is a pretty useful utility, it should live somewhere that other components can access it
* Rather than hard coding words in a specific language, it might be more useful to make those strings arguments of the `formatDate` function (maybe with default args using the english words you have in this?)
* For even more flexibility, you could add a `locale` argument too. 
* You could memoize this function. I'd imagine that there are going to be many of these on a page.

Reply:
I thought about that. I wasn't sure what kind of options/use cases would be needed. I'll tinker around and add some robustness to it. And move it.

Reply:
Also, I'm not sure about the memoization here, since pretty much every single message will have a unique timestamp, unless I'm missing something.

Reply:
Those messages will re-render as you get more messages. If the timestamp is memoized then the browser doesn't have to re-calculate the formatted date on each render.

Reply:
Oh, right! Nice call.